hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/infra/HevmTransactionFactory.java

```diff
@@ -278,7 +278,7 @@ public HederaEvmTransaction fromContractTxException(
             throw new HandleException(hydratedEthTxData.status());
         }
         final var ethTxData = requireNonNull(hydratedEthTxData.ethTxData());
-        validateTrue(ethTxData.matchesChainId(asChainIdBytes(contractsConfig.chainId())), WRONG_CHAIN_ID);
+        validateTrue(ethTxData.matchesChainId(Integers.toBytes(contractsConfig.chainId())), WRONG_CHAIN_ID);
```

Comment:
I'd like a better explanation of why this will not throw a `HandleException` with `WRONG_CHAIN_ID` than simply a thumbs down.  In your test in LeakyEthereumTestsSuite you had to override the `contracts.chain_id` property and set the value to 0 in order for your test `legacyUnprotectedEtxBeforeEIP155` to run.  That property will not be set to 0 in mainnet.  With this change will legacy transactions work without changing `contracts.chain_id`?

Reply:
Thanks for pointing that out! Added another test as well as additional handle in `EthTxData.matchesChainId()` method

Reply:
Nice! Update looks good to me.